sktime/utils/_testing/scenarios.py

```diff
@@ -0,0 +1,200 @@
+# -*- coding: utf-8 -*-
+"""Testing utility to play back usage scenarios for estimators.
+
+Contains TestScenario class which applies method/args subsequently
+"""
+
+__author__ = ["fkiraly"]
+
+__all__ = ["TestScenario"]
+
+
+from copy import deepcopy
+from inspect import isclass
+
+import numpy as np
+
+
+class TestScenario:
+    """Class to run pre-defined method execution scenarios for objects.
+
+    Parameters
+    ----------
+    args : dict of dict, default = None
+        dict of argument dicts to be used in methods
+        names for keys need not equal names of methods these are used in
+            but scripted method will look at key with same name as default
+        must be passed to construtor, or set in a child class
+    default_method_sequence : list of str, default = None
+        optional, if given, default method sequence to use in "run"
+        if not provided, "run" must alwayss be passed a method_sequence
+    default_arg_sequence : list of str, default = None
+        sequence of keys for keyword argument dicts to be used
+        names for keys need not equal names of methods
+
+    Methods
+    -------
+    run(obj, args=None, default_method_sequence=None)
+        Run a call(args) scenario on obj, and retrieve method outputs.
+    is_applicable(obj)
+        Check whether scenario is applicable to obj.
+    """
+
+    def __init__(
+        self, args=None, default_method_sequence=None, default_arg_sequence=None
+    ):
+        if default_method_sequence is not None:
+            self.default_method_sequence = _check_list_of_str(default_method_sequence)
+        if default_arg_sequence is not None:
+            self.default_arg_sequence = _check_list_of_str(default_arg_sequence)
+        if args is not None:
+            self.args = _check_dict_of_dict(args)
+        else:
+            if not hasattr(self, args):
+                raise RuntimeError(
+                    "args must either be given to __init__ or set in a child class"
+                )
+            _check_dict_of_dict(self.args)
+
+    def run(self, obj, method_sequence=None, arg_sequence=None, return_all=False):
```

Comment:
not clear to me why we allow overwriting of `method_sequence` and `arg_sequences`, but not `args`. what was the reason for this particular choice?

Note my instinct is that we do not need any over-writing and that we would create a new TestScenario object when we need to change something. but you must have had a reason to do things that way you did!

Reply:
if you would pass all three, would it not be easier, in terms of writing code to just run the methods explicitly?

The key (intended) benefit of this design is to allow appling the same sequence of input to the same sequence of methods across many estimators.

The ability to choose `method_sequence` is for cases where you may want very similar sequences with the same data, e.g., `fit-predict` vs `fit-predict-update` or `fit-predict-update-predict`.

Reply:
resolving, moving general design discussion into main area, rather than in multiple conversation